app/src/main/java/com/kickstarter/ui/activities/ProjectActivity.kt

```diff
@@ -144,9 +176,75 @@ class ProjectActivity : BaseActivity<ProjectViewModel.ViewModel>() {
         this.project_recycler_view.adapter = null
     }
 
-    private fun renderProject(project: Project, configCountry: String) {
-        this.adapter.takeProject(project, configCountry)
-        ProjectUtils.setActionButton(project, this.back_project_button, this.manage_pledge_button, this.view_pledge_button)
+    private fun animateRewards(expanded: Boolean) {
+        this.showRewardsFragment.removeAllUpdateListeners()
+        this.showRewardsFragment.addUpdateListener { valueAnim ->
+            val initialRadius = resources.getDimensionPixelSize(R.dimen.fab_radius).toFloat()
+            val radius = initialRadius * if (expanded) 1 - valueAnim.animatedFraction else valueAnim.animatedFraction
+            this.rewards_container.radius = radius
+        }
+
+        val durationTransition = AutoTransition()
+        durationTransition.duration = animDuration
+        TransitionManager.beginDelayedTransition(root, durationTransition)
+
+        val rewardAlphaSet = AnimatorSet()
+        this.showRewardsFragment.target = if (!expanded) native_back_this_project_button else pledge_container
+        this.hideRewardsFragment.target = if (!expanded) pledge_container else native_back_this_project_button
+        rewardAlphaSet.playTogether(this.showRewardsFragment, this.hideRewardsFragment)
+        rewardAlphaSet.duration = animDuration
+
+        rewardAlphaSet.addListener(object : Animator.AnimatorListener {
+            override fun onAnimationRepeat(animation: Animator?) {}
+            override fun onAnimationCancel(animation: Animator?) {}
+
+            override fun onAnimationEnd(animation: Animator?) {
+                if (expanded) {
+                    native_back_this_project_button.visibility = View.GONE
+                }
+            }
+
+            override fun onAnimationStart(animation: Animator?) {
+                setRewardConstraints(expanded)
+                if (!expanded) {
+                    native_back_this_project_button.visibility = View.VISIBLE
+                }
+            }
+        })
+
+        rewardAlphaSet.start()
+    }
+
+    private fun displayProjectAndRewards(projectCountryAndNativeCheckout: Pair<Pair<Project, String>, Boolean>) {
+        this.renderProject(projectCountryAndNativeCheckout.first.first, projectCountryAndNativeCheckout.first.second, projectCountryAndNativeCheckout.second)
```

Comment:
can we please make objects out of these since it's hard to read?
like `val project = projectCountryAndNativeCheckout.first.first` etc